proxy/src/transport/tls/config.rs

```diff
@@ -110,27 +169,70 @@ impl CommonSettings {
             })
     }
 
-    /// Stream changes to the files described by this `CommonSettings`.
-    ///
-    /// This will poll the filesystem for changes to the files at the paths
-    /// described by this `CommonSettings` every `interval`, and attempt to
-    /// load a new `CommonConfig` from the files again after each change.
-    ///
-    /// The returned stream consists of each subsequent successfully loaded
-    /// `CommonSettings` after each change. If the settings could not be
-    /// reloaded (i.e., they were malformed), nothing is sent.
-    ///
-    /// TODO: On Linux, this should be replaced with an `inotify` watch when
-    ///       available.
-    pub fn stream_changes(self, interval: Duration)
-        -> impl Stream<Item = CommonConfig, Error = ()>
+    #[cfg(target_os = "linux")]
+    fn stream_changes_inotify(&self)
+        -> Result<impl Stream<Item = (), Error = ()>, Error>
     {
-        self.change_timestamps(interval)
-            .filter_map(move |_|
-                CommonConfig::load_from_disk(&self)
-                    .map_err(|e| warn!("error reloading TLS config: {:?}, falling back", e))
-                    .ok()
-            )
+        use inotify::{Inotify, WatchMask};
+        // If we used a less broad watch mask, we could avoid reloading the
+        // certs multiple times when k8s modifies a ConfigMap or Secret (a
+        // multi-step process that we see as a series of CREATE, MOVED_TO,
+        // MOVED_FROM, and DELETE events). However, this may not catch changes
+        // if the files we're watching *don't* live in a k8s ConfigMap/Secret,
+        // and I think false positives are much less harmful here than false
+        // negatives.
```

Comment:
What I get out of the comment is that there's another way to do it that we're not doing... What do I need to understand from this comment to to understand the code that's here?